pkgs/development/compilers/go/1.11.nix

```diff
@@ -133,12 +133,12 @@ stdenv.mkDerivation rec {
 
   GOOS = if stdenv.isDarwin then "darwin" else "linux";
   GOARCH = if stdenv.isDarwin then "amd64"
-           else if stdenv.targetPlatform.isi686 then "386"
```

Comment:
This should be correct. GOARCH is the architecture that the go compiler will produce code /for/. targetPlatform is used correctly here. The stdenv.isDarwin stuff looks wrong though. We should be able to use ```targetPlatform.parsed.kernel.name``` for GOOS and a simple map for GOARCH:

```
GOARCH = {
  "i686" = "386";
  "x86_64" = "amd64";
  "aarch64" = "arm64";
  "arm" = "arm";
  "armv5tel" = "arm";
  "armv6l" = "arm";
  "armv7l" = "arm";
}.${stdenv.targetPlatform.parsed.cpu.name} or (throw "Unsupported system");
```

Reply:
The go compiler, after it's built, is able to produce code for any platform by changing those environment variables. So you'd set those appropriately for the host platform when doing a cross compile of a go package like terraform. But when building the go compiler itself, setting it to anything other than the build platform results in "invalid instruction" build errors.

Reply:
Makes sense, but that restructuring that Matt wrote (the "switch-case") is also a good idea.

Reply:
Ok sounds good!

Reply:
Updated to use the switch-case version